### ISSUE TITLE
Scribble crashes my game!

### ISSUE THREAD

**Issue by DelayedVictory** [JujuAdams/Scribble#512]
I've implemented scribble in my game, but recently people have started to report Scribble-related crashes that I cannot seem to fix. I cannot reproduce the issue on my end so I was hoping to get some answers here. I'm consistently getting 2 crash reports.

Crash report #1:
For context - the object referenced in the error (obj_gate) does not contain any scribble code at all. Neither does it have any parents or child objects containing scribble code.
```
___
############################################################################################
ERROR in
action number 1
of Create Event
for object obj_gate:

DoAdd :2: undefined value
at gml_Script_scribble_tick
############################################################################################
gmlScriptscribble_tick (line -1)
gml_Script_anon@950@scribbleinitializescribble_system
```

Crash report #2:
```
___
############################################################################################
ERROR in
action number 1
of Draw Event
for object obj_header_money:

Variable <unknown_object>.__flushed(100103, -2147483648) cannot be resolved.
at gml_Script_scribble
############################################################################################
gml_Script_scribble (line -1)
gml_Object_obj_header_money_Draw_64
```

The object referenced in the error here does contain the following scribble code:
```
var _money = string(floor(money));
scribble("[spr_money]" + _money).scale(gui_text_scale).align(halign, valign).blend(color, 1).draw(gui_bbox_left + gui_text_margin, gui_y);
```

**Comment by DelayedVictory** [comment]
After giving it more thought it seems the errors are being reported by players shortly after updating the GameMaker Runtime to 11.1.160.
For more context - my game has a consistent playerbase of 1k+ daily users and the issue hasn't been reported before October 17th. Reverting runtime now to see if it still comes up.

**Comment by JujuAdams** [comment]
What version of Scribble are you on? What platforms are you encountering the bug on? Are you distributing your game using YYC? Do you use any manual garbage collection?

Somewhay unlikely to be Scribble itself considering the highly weird nature of these crashes and their recent occurrence. For example, `__flushed` is an intrinsic property of Scribble text elements and the text element cache only contains text elements so that variable must always exist for every member of the contents of that array.

From experience solving weird crashes in the past I'd guess that this is either:

1. 2023.11 broke something inside the garbage collector
2. Some sort of global token (globalvar, macro, script, perhaps enum) is interfering with Scribble's operation
3. Rarely, a bug in YYC

It is going to be hard to track this down properly due to lack of repro. However, it looks like an invalid text element reference is being found inside Scribble's internal cache, either due to it being invalid to begin with or it being corrupted, or a bug in the garbage collector breaking something internally.

Quick word on that strange Create event error message - `__scribble_tick()` runs in a time source and that might be throwing the event name off.

I have some suggestions as to how to track down this bug. First thing is to set up some try...catch blocks in known sensitive areas to try to catch the issue as it happens. You can then mitigate the crash (hopefully it doesn't snowball!) and, more importantly, dump state to disk for analysis later. The real power move here is to set up a bug tracking service to automatically collect bug reports and state dumps.

If you do decide to implement try...catch blocks, please make multiple try...
catch blocks inside `__scribble_tick()`. It is not immediately clear to be exactly what part of it is crashing.

What state should you dump? I'd start with the contents of Scribble's internal caches. That'll tell us a lot about what Scribble thinks is happening. I'd also store any relevant game state too - what room the player is in, where they are in the room etc.  You'll need to use your kmowledge of your own game to figure out what's going on.

Let me know when you have some more data (exact repro obviously preferred!) and we can talk further.

**Comment by DelayedVictory** [comment]
Just wanted to follow this up to say that I reverted GameMaker Runtime version and the issue has since not come up again - so I assume there's an issue with Scribble and GameMaker Runtime 11.1.160.

**Comment by DelayedVictory** [comment]
Also to answer your questions:
What version of Scribble are you on? Latest one
What platforms are you encountering the bug on? PC (Steam)
Are you distributing your game using YYC? Yes
Do you use any manual garbage collection? No, I just have users send me crash errors

**Comment by JujuAdams** [comment]
Did you make any further progress on this?

**Comment by DelayedVictory** [comment]
> Did you make any further progress on this?

Thanks for checking back. I have managed to fix the issue, but I don't understand the logic behind it.

This piece of code was causing the game to crash when var _money was between 1.204.000 and 1.210.000 for which I have absolutely no logical explanation:
```
var _money = string(floor(money));
scribble("[spr_money]" + _money).scale(gui_text_scale).align(halign, valign).blend(color, 1).draw(gui_bbox_left + gui_text_margin, gui_y);
```

I have tried different things for hours but eventually managed to 'fix' the crash by adding a space:
`"[spr_money]"` > `"[spr_money] "`

I have not received any crash reports since then, so I'm confident this solved the issue. I don't understand what difference the space makes though.
